app/workers/concerns/dump_worker.rb

```diff
@@ -128,13 +128,7 @@ def get_resource
     @resource_type.camelize.constantize.find(@resource_id)
   end
 
-  def read_from_database
-    if Panoptes.flipper.enabled?("dump_data_from_read_slave")
-      Slavery.on_slave do
-        yield
-      end
-    else
-      yield
-    end
+  def read_from_database(&block)
+    DatabaseReplica.read("dump_data_from_read_slave", &block)
```

Comment:
this is costlier than using a block as it converts the block to a proc but it keeps the code cleaner.